test/Chainweb/Test/Mempool/Socket.hs

```diff
@@ -8,14 +8,36 @@ import Control.Concurrent
 import Control.Exception
 import qualified Data.Pool as Pool
 ------------------------------------------------------------------------------
+import Chainweb.BlockHeaderDB
+import Chainweb.Graph (singletonChainGraph)
 import Chainweb.Mempool.InMem (InMemConfig(..))
 import qualified Chainweb.Mempool.InMem as InMem
 import Chainweb.Mempool.Mempool
+import Chainweb.Payload.PayloadStore
 import qualified Chainweb.Mempool.Socket as M
 import Chainweb.Test.Mempool (MempoolWithFunc(..))
 import qualified Chainweb.Test.Mempool
+import Chainweb.Test.Utils (toyChainId)
 import Chainweb.Utils (Codec(..))
+import Chainweb.Version
+import Data.CAS.RocksDB
+
 ------------------------------------------------------------------------------
+tests :: TestTree
+tests = withResource (newPool cfg) Pool.destroyAllResources $
```

Comment:
This resource allocation should probably happen higher up.

Reply:
I started to change this, but I'm going to have to put it on the list for another PR.  I didn't write this code and its a bit complicated to pull out the various IO resources.